### PR TITLE
MDC fixes

### LINKED ISSUE #117: V15 MDC Fixes
Here is a branch where I fixed up all the demo issues.  All contrast colors work as expected. Palette variables loading up properly now. This is a pure MDC implementation no legacy components. Previous legacy fixes have been removed.  I used a completely different strategy to determine when rgb or no-rgb variables should be used. 

https://github.com/pedrojrivera/angular-material-css-vars/tree/feature/v15-mdc

### PR DESCRIPTION
**Pull request by json-derulo**
# Description

* Resolve issues with MDC components (credits: @pedrojrivera)
* BREAKING: Drop support for legacy components
* BREAKING: Drop support for background / foreground color variables

## Issues Resolved

Fixes #117 

## Check List

- [ ] New functionality includes testing.
- [ ] New functionality has been documented in the README if applicable.

### CODE REVIEW

File: projects/material-css-vars/src/lib/_mat-lib-overwrites.scss [hunk 1]
```diff
@@ -1,143 +1,377 @@
...
-
-  @return map-get($palette, $hue);
-}
-
-// apply additional overwrites
+// ---------------------------
+// MDC Overwrites
+// ---------------------------
 @mixin other-overwrites {
-  @include _mat-date-picker-content-overwrites();
```
Comment (json-derulo) [inline projects/material-css-vars/src/lib/_mat-lib-overwrites.scss:16]: The date picker was not affected by the MDC update, need to check whether still need this overwrites

**Comment by json-derulo** [comment]
I think I found two more issues:
* Form field background colors are slightly off
* Disabled chips have the wrong background color

**Review by json-derulo — COMMENTED** [review]
I think I found two more issues:
* Form field background colors are slightly off
* Disabled chips have the wrong background color

**Review by nekkon — COMMENTED** [review]
Would be great to have this reviewed/merged.

**Pull request merged**